superset/datasets/commands/importers/v1/utils.py

```diff
@@ -136,7 +136,7 @@ def load_data(
             df[column_name] = pd.to_datetime(df[column_name])
 
     # reuse session when loading data if possible, to make import atomic
-    if example_database.sqlalchemy_uri == get_main_database().sqlalchemy_uri:
+    if example_database.sqlalchemy_uri == get_example_database().sqlalchemy_uri:
```

Comment:
@dpgaspar this condition will always be true, and causes the example data to be loaded into the main database instead of the examples databases (because it always reuses `session.connection()`.